### DISCUSSION TITLE
How to fetch data only on the server-side (during generate) when using SSG?

### DISCUSSION THREAD

**Discussion by gjfLeo** [nuxt/nuxt#22396] [Questions]
I want to create a route (i.e. `/article/1`) that only fetch data on server-side (during `nuxi generate`). Is this possible? 

I have some `/server/api` in nuxt app, and its result would never change. Now I directly use `useFetch` in `setup`. In SSG generated site, it works when visit `/article` directly (no request on client-side). However, when you reach the route via other page (i.e. from `/home` to `/article/1`), there would be a request on client-side (and failed of course). How can I always use generated site with server-fetched data?

**Comment by jwhazel** [comment]
If I understand your question, `/article/1` is a dynamic route and that's why it doesn't get picked up during generate. If so you need to give Nitro a hint about the available routes. In nuxt.config.ts:
```
hooks: {
    async "nitro:config"(nitro: any) {
          nitro.prerender.routes = ['/article/1']; // add any other dynamic routes you need here
    },
  },
```
